src/universal/modules/userDashboard/containers/UserSettings/UserSettingsContainer.js

```diff
@@ -76,7 +88,56 @@ export default class UserSettingsContainer extends Component {
       router.push(nextPage);
     }
     untouch('preferredName');
-  };
+  }
+
+  uploadPicture(pictureFile) {
```

Comment:
Yeah, you've got it! I did a bunch of reading before I elected this implementation. Those are the two big choices: 1) send the file to the server, the server puts it up on S3, 2) the client requests a signed URL from the server, posts the file to S3 itself, then notifies the server.

In the end, the latency for multiple small server transactions is _probably_ less than sending the file twice (once to Parabol, then from Parabol to S3). This gets truer and truer the larger and larger the asset is. All of a sudden, that 250kB avatar image feels like a 500kB avatar image because it needs to be sent twice.

If we decide, at some point, to ever add file attachments to Project cards or whatnot we can do so now fairly easily, and at low cost in terms of bandwidth or storage. However, we don't always have to use the pre-signed URL strategy. If we find that we prefer uploading a type of assets to the Parabol server, then to S3, we can totally do so.

I'm happy for now, but I'm always happy to massage it in the future.